### PR TITLE
feat: add compare validator

### PR DESCRIPTION
**Pull request by kernel-io**
feat: added number validator supporting validation of arguments and attributes, all the standard validators are included i.e: greater_than, less_than, less_than_or_equal_to, greater_than_or_equal_to
feat: added two new functions in changeset, `&get_argument_or_attribute/2` and `&fetch_argument_or_change/2`
fix: give `present` validation access to arguments also, not just arguments

added some tests

### CODE REVIEW

File: lib/ash/resource/validation/number.ex [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      _ ->
+        :ok
+    end
+  end
+
+  defp invalid_attribute_error(opts, message \\ nil) do
+    {:error,
+     InvalidAttribute.exception(
+       field: opts[:attribute],
+       message: message || opts[:message],
```
Comment (kernel-io) [inline lib/ash/resource/validation/number.ex:109]: causing the dialyzer to rightly complain

Comment (zachdaniel) [reply lib/ash/resource/validation/number.ex:109]: Yeah, you want `opts[:message] || message` here right?

**Comment by zachdaniel** [comment]
Can you format the code with `mix format`, take a look at the compiler warning and then if you fix the `message || opts[:message]` thing it should be good to go 👍🏻 great work!

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀w

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**